src/metabase/channel/params.clj

```diff
@@ -0,0 +1,27 @@
+(ns metabase.channel.params
+  (:require
+   [clojure.string :as str]
+   [metabase.driver.common.parameters :as params]
+   [metabase.driver.common.parameters.parse :as params.parse]))
+
+(defn- param-name->path
+  [param-name]
+  (->> (str/split param-name #"\.")
+       (mapv keyword)))
+
+(defn substitute-params
+  "Substitute parameters in text with values from context.
+
+  Params are specified using mustache syntax, e.g. {{param}}."
+  [text context]
+  ;; NOTE: in case the syntax involves, consider using the mustache syntax and use stencil for substitution
```

Comment:
I see stencile is ancient, I wonder whether we shouldn't consider moving to this "newer"[^1] variant, which is also the "official" one (linked on the Mustache website).

[^1]: 2 years old years since last commit, versus 9.